### PR TITLE
refactor: update logging and error messages for consistency

### PR DESCRIPTION
**Pull request by kruskall**
format and make sure zap field enrichment and logged messages are consistent

### CODE REVIEW

File: kafka/consumer.go [hunk 1]
```diff
@@ -228,6 +231,7 @@ func (c *Consumer) fetch(ctx context.Context) error {
 			c.cfg.Logger.Error("unable to process event",
 				zap.Error(err),
 				zap.String("topic", msg.Topic),
+				zap.ByteString("message.value", msg.Value),
```
Comment (kruskall) [inline kafka/consumer.go:234]: **question**: logging of headers and message values seem to differ wildly. When should they be logged ? Does it make sense to always log them ?

File: kafka/consumer.go [hunk 2]
```diff
@@ -228,6 +231,7 @@ func (c *Consumer) fetch(ctx context.Context) error {
...
 				zap.Int64("offset", msg.Offset),
 				zap.Int32("partition", msg.Partition),
 				zap.Any("headers", meta),
```
Comment (kruskall) [inline kafka/consumer.go:237]: **question**: should we use a more general name for the log field ? `metadata` maybe ?

File: kafka/consumer.go [hunk 3]
```diff
@@ -180,8 +180,11 @@ func (c *Consumer) fetch(ctx context.Context) error {
...
-	if fetches.IsClientClosed() || errors.Is(fetches.Err0(), context.Canceled) {
-		return context.Canceled // Client closed or context cancelled.
+	if fetches.IsClientClosed() {
+		return fmt.Errorf("client is closed: %w", context.Canceled)
+	}
+	if errors.Is(fetches.Err0(), context.Canceled) {
+		return fmt.Errorf("context canceled: %w", fetches.Err0())
 	}
 	switch c.cfg.Delivery {
 	case apmqueue.AtLeastOnceDeliveryType:
```
Comment (kruskall) [inline kafka/consumer.go:190]: **question**: we are not checking for error when committing uncommited offsets. Should we log an error or/and return the error to the caller ?

File: kafka/consumer.go [hunk 4]
```diff
@@ -109,7 +109,7 @@ type Consumer struct {
 // NewConsumer creates a new instance of a Consumer.
 func NewConsumer(cfg ConsumerConfig) (*Consumer, error) {
 	if err := cfg.Validate(); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("invalid consumer config: %w", err)
```
Comment (marclop) [inline kafka/consumer.go:112]:
```suggestion
		return nil, fmt.Errorf("kafka: invalid consumer config: %w", err)
```

File: kafka/consumer.go (hunk 3 above)
Comment (marclop) [reply kafka/consumer.go:190]: No need, since this function is running on a loop. 

If the offsets are committed back to Kafka, the events will be re-processed once the processing timeout is hit, adhering to the desired at least once delivery.

File: kafka/consumer.go (hunk 1 above)
Comment (marclop) [reply kafka/consumer.go:234]: I added the message value in the log above since that error occurs on decoding a message. Processing the batch may happen for any reason and it seemed to me that it wasn't necessary to log the raw `message.value`. I still think it's best to not log the message.value wherever possible.

File: kafka/producer.go [hunk 5]
```diff
@@ -181,6 +181,10 @@ func (p *Producer) ProcessBatch(ctx context.Context, batch *model.Batch) error {
 				p.cfg.Logger.Error("failed producing message",
 					zap.Error(err),
 					zap.String("topic", msg.Topic),
+					zap.ByteString("message.value", msg.Value),
```
Comment (marclop) [inline kafka/producer.go:184]: I'd like to avoid logging the `message.value` unless necessary for debugging.
```suggestion
```

File: pubsublite/consumer.go [hunk 6]
```diff
@@ -133,7 +133,7 @@ type Consumer struct {
 // NewConsumer creates a new consumer instance for a single subscription.
 func NewConsumer(ctx context.Context, cfg ConsumerConfig) (*Consumer, error) {
 	if err := cfg.Validate(); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("invalid consumer config: %w", err)
```
Comment (marclop) [inline pubsublite/consumer.go:136]:
```suggestion
		return nil, fmt.Errorf("pubsublite: invalid consumer config: %w", err)
```

File: pubsublite/consumer.go [hunk 7]
```diff
@@ -253,8 +254,10 @@ func (c consumer) processMessage(ctx context.Context, msg *pubsub.Message) {
 		partition, offset := partitionOffset(msg.ID)
 		c.logger.Error("unable to process event",
 			zap.Error(err),
+			zap.ByteString("message.value", msg.Data),
```
Comment (marclop) [inline pubsublite/consumer.go:257]:
```suggestion
```

File: pubsublite/producer.go [hunk 8]
```diff
@@ -96,7 +96,7 @@ type Producer struct {
 // NewProducer creates a new PubSub Lite producer for a single project.
 func NewProducer(ctx context.Context, cfg ProducerConfig) (*Producer, error) {
 	if err := cfg.Validate(); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("invalid producer config: %w", err)
```
Comment (marclop) [inline pubsublite/producer.go:99]:
```suggestion
		return nil, fmt.Errorf("pubsublite: invalid producer config: %w", err)
```

File: pubsublite/producer.go [hunk 9]
```diff
@@ -161,17 +161,23 @@ func (p *Producer) ProcessBatch(ctx context.Context, batch *model.Batch) error {
...
-	for _, res := range responses {
-		if serverID, err := res.Get(ctx); err != nil {
-			p.cfg.Logger.Error("failed producing message",
-				zap.Error(err),
-				zap.String("server_id", serverID),
-			)
-		}
+		result := producer.Publish(ctx, &msg)
+		wg.Add(1)
+		go func() {
```
Comment (marclop) [inline pubsublite/producer.go:166]: This PR is about updating logging and error messages for consistency, this seems a much greater change than that. Could you revert these changes, please?

**Review by marclop — CHANGES_REQUESTED** [review]

File: kafka/consumer.go (hunk 2 above)
Comment (marclop) [reply kafka/consumer.go:237]: ideally both Kafka and pubsublite should have the same field. Either headers or metadata seem fine to me.

File: kafka/consumer.go (hunk 3 above)
Comment (kruskall) [reply kafka/consumer.go:190]: Oh I see, thanks for explanation! :bow:

File: kafka/consumer.go (hunk 1 above)
Comment (kruskall) [reply kafka/consumer.go]: Oh I see, I've updated the code to remove the message.value during processing errors.

Should the headers also be removed ? Are they valuable ?

File: pubsublite/producer.go (hunk 9 above)
Comment (kruskall) [reply pubsublite/producer.go]: Ah, this was to avoid storing tmp variables to retrieve topics and headers later on. I've changed it to a tmp variable

File: kafka/consumer.go (hunk 2 above)
Comment (kruskall) [reply kafka/consumer.go]: Let's keep headers then!

File: kafka/consumer.go (hunk 1 above)
Comment (marclop) [reply kafka/consumer.go]: I think it's fine to log the headers, and may be useful to be able to identify failing patterns for given metadata.

**Review by marclop — APPROVED** [review]
LGTM. Thank you!
